### PR TITLE
Codegen new intrinsic `raw_eq`

### LINKED ISSUE #330: Codegen failure: Unimplemented `raw_eq` intrinsic
The Rust team has introduced a new intrinsic `raw_eq` [here](https://github.com/rust-lang/rust/commit/ee86f96ba176f598d64dc9f3bb7e074d5b8b86b6). Failure to codegen is blocking this week's rebase in #292.

### PR DESCRIPTION
**Pull request by adpaco-aws**
### Description of changes: 

This allows us to codegen the intrinsic `raw_eq` which has been introduced in the latest rebase. See #292 for the report.

### Resolved issues:

Resolves #330 
Unblocks #292 

### Call-outs:

Codegen can be improved to avoid calling `memcmp` in some cases.

### Testing:

* How is this change tested? Existing regression plus one new test for this intrinsic.

* Is this a refactor change? No.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Review by chinmaydd — APPROVED** [review]
LGTM!

File: compiler/rustc_codegen_llvm/src/gotoc/intrinsic.rs [hunk 1]
```diff
@@ -666,6 +667,33 @@ impl<'tcx> GotocCtx<'tcx> {
...
     }
 
+    // `raw_eq` determines whether the raw bytes of two values are equal.
+    // https://stdrs.dev/nightly/x86_64-pc-windows-gnu/std/intrinsics/fn.raw_eq.html
+    //
+    // The implementation below calls `memcmp` and returns equal if the result is zero.
+    //
+    // TODO: Handle more cases in this intrinsic by looking into the parameters' layouts.
+    // TODO: Fix soundness issues in this intrinsic. It's UB to call `raw_eq` if any of
+    // the bytes in the first or second arguments are uninitialized.
```
Comment (tedinski) [inline compiler/rustc_codegen_llvm/src/gotoc/intrinsic.rs:677]: Are there things we could `assert` on here? Or for the first TODO would it already crash in the unhandled cases?

**Review by tedinski — APPROVED** [review]

**Review by vecchiot-aws — APPROVED** [review]
:+1:

File: compiler/rustc_codegen_llvm/src/gotoc/intrinsic.rs (hunk 1 above)
Comment (adpaco-aws) [reply compiler/rustc_codegen_llvm/src/gotoc/intrinsic.rs:677]: The soundness check has to come in before the rest of the code. Then you can decide based on layouts and finally use `memcmp` for the general case. The cranelift backend does [this](https://github.com/rust-lang/rust/commit/ee86f96ba176f598d64dc9f3bb7e074d5b8b86b6#diff-2b7a0a08fe6fe065be4b1836cb71d7d09512145dbc1d4093816fa3bbb2768136).

**Pull request merged**